l10n_nl_bsn/models/res_partner.py

```diff
@@ -0,0 +1,68 @@
+# Copyright 2016-2018 Onestein (<https://www.onestein.eu>)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+import logging
+
+from odoo import api, fields, models, _
+
+_logger = logging.getLogger(__name__)
+try:
+    from stdnum.nl import bsn
+except ImportError:
+    _logger.debug('Cannot `import stdnum.nl.bsn`.')
+
+
+class ResPartner(models.Model):
+    _inherit = 'res.partner'
+
+    bsn_number = fields.Char(
+        string='BSN',
+        groups='hr.group_hr_user'
+    )
+
+    @api.multi
+    @api.onchange('bsn_number')
+    def onchange_bsn_number(self):
+        warning = {}
+        for partner in self:
+            if partner.bsn_number:
+                # properly format the entered BSN
+                partner.bsn_number = bsn.format(partner.bsn_number)
+
+                # check is valid, otherwise display a warning
+                warning = partner._warn_bsn_invalid()
+
+                # search for another partner with the same BSN
+                args = [('bsn_number', '=', partner.bsn_number),
+                        ('name', '!=', partner.name)]
+
+                # refine search in case of multicompany setting
+                if partner.company_id:
+                    args += [('company_id', '=', partner.company_id.id)]
+                other_partner = self.search(args, limit=1)
+
+                # if another partner exists, display a warning
+                if other_partner:
+                    warning = other_partner._warn_bsn_existing()
+        return {'warning': warning, }
```

Comment:
If the BSN is not well formatted AND is already in use, you'd get a warning only for the latter, as the former value of the variable would be overwritten.
Is that the wished behavior? Wouldn't be better to have both issues signaled at once?

Reply:
I assumed that the warning "already in use" has the priority over the "not well formatted" one. You would likely check the "already in use" case first, in case both conditions happen.

Reply:
If it's the wished behavior that's fine by me. Thanks